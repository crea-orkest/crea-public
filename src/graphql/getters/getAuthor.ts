import type { RequestInit } from 'next/dist/server/web/spec-extension/request'
import { authorFormatter } from '../formatters/authorFormatter'
import { client } from '../gqlClient'
import {
  GetAuthorDocument,
  type GetAuthorQuery,
  type GetAuthorQueryVariables,
} from '../generated/graphql'

export const getAuthor = async (
  { id }: GetAuthorQueryVariables,
  fetchOptions?: RequestInit
) => {
  try {
    const { data, error } = await client.query<
      GetAuthorQuery,
      GetAuthorQueryVariables
    >(GetAuthorDocument, { id }, { fetchOptions })

    return {
      data: data ? authorFormatter(data) : null,
      error,
    }
  } catch (error) {
    if (error instanceof Error) console.log(error.message)

    return { data: null, error }
  }
}
