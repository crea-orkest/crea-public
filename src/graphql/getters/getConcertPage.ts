import { client } from '../gqlClient'
import {
  GetConcertPageDocument,
  type GetConcertPageQuery,
  type GetConcertPageQueryVariables,
} from 'graphql/generated/graphql'

export const getConcertPage = async ({
  slug,
}: GetConcertPageQueryVariables) => {
  try {
    const { data, error } = await client.query<
      GetConcertPageQuery,
      GetConcertPageQueryVariables
    >(GetConcertPageDocument, { slug })
    return {
      data: data?.concert ?? null,
      error,
    }
  } catch (error) {
    if (error instanceof Error) console.log(error.message)
    return { data: null, error }
  }
}
