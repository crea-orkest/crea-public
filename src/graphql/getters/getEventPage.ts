import { client } from 'graphql/gqlClient'
import { eventFormatter } from 'graphql/formatters/eventFormatter'
import {
  GetEventPageDocument,
  type GetEventPageQuery,
  type GetEventPageQueryVariables,
} from 'graphql/generated/graphql'

export const getEventPage = async ({ slug }: GetEventPageQueryVariables) => {
  try {
    const { data, error } = await client.query<
      GetEventPageQuery,
      GetEventPageQueryVariables
    >(GetEventPageDocument, { slug })
    return {
      data: data?.concert ? eventFormatter(data.concert) : null,
      error,
    }
  } catch (error) {
    if (error instanceof Error) console.log(error.message)
    return { data: null, error }
  }
}
