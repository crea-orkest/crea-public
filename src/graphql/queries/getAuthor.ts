import gql from "graphql-tag";

import { author } from "../fragments/author";

export const GET_AUTHORS = gql`
  query getAuthor($id: ItemId) {
    person(filter: { id: { eq: $id } }) {
      ...author
    }
  }
  ${author}
`;
