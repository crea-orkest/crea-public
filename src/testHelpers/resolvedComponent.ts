export const resolvedComponent = async <T>(
  Component: Function,
  props: T
): Promise<() => JSX.Element> => {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
};
