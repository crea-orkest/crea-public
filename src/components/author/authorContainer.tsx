import React from 'react'
import { AuthorView } from './authorView'
import { getAuthor } from '../../graphql/getters/getAuthor'

export interface Props {
  id: string
}

export const AuthorContainer: React.FC<Props> = async ({ id }) => {
  const { data } = await getAuthor({ id })
  if (!data) return null

  return <AuthorView data={data} />
}
