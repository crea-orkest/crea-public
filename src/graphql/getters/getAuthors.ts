import { GET_AUTHORS } from "graphql/queries/getAuthors";
import { authorsFormatter } from "../formatters/authorsFormatter";
import {
  GetAuthorsQuery,
  GetAuthorsQueryVariables,
} from "../generated/graphql";
import { getClient } from "../gqlClient";

export const getAuthors = async ({
  first,
  skip,
  order,
}: GetAuthorsQueryVariables) => {
  const client = getClient();

  try {
    const { data, loading, errors } = await client.query<GetAuthorsQuery>({
      query: GET_AUTHORS,
      variables: { first, skip, order },
    });
    return { data: authorsFormatter(data), loading, errors };
  } catch (errors) {
    if (errors instanceof Error) console.log(errors.message);

    return { data: null, loading: false, errors };
  }
};
