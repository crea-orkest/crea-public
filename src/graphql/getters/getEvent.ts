import { getClient } from "../gqlClient";
import {
  GetConcertQuery,
  GetConcertQueryVariables,
} from "../generated/graphql";
import { eventFormatter } from "graphql/formatters/eventFormatter";
import { GET_CONCERT } from "graphql/queries/getConcert";

export const getEvent = async ({ id }: GetConcertQueryVariables) => {
  const client = getClient();
  try {
    const { data, loading, errors } = await client.query<GetConcertQuery>({
      query: GET_CONCERT,
      variables: { id },
    });

    return { data: eventFormatter(data), loading, errors };
  } catch (errors) {
    if (errors instanceof Error) console.log(errors.message);
    return { data: null, loading: false, errors };
  }
};
