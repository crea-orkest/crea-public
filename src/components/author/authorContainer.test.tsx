import { AuthorContainer } from './authorContainer'
import type { Props } from './authorContainer'
import React from 'react'
import { getAuthor } from '../../graphql/getters/getAuthor'
import { resolvedComponent } from 'utils/testHelpers/resolvedComponent'
import { render, screen } from '@testing-library/react'

jest.mock('../../graphql/getters/getAuthor', () => {
  const originalModule = jest.requireActual('../../graphql/getters/getAuthor')
  return {
    __esModule: true,
    ...originalModule,
    getAuthor: jest.fn(),
  }
})

const getAuthorMock = jest.mocked(getAuthor)

describe('Author component', () => {
  it('shows all the data', async () => {
    getAuthorMock.mockResolvedValueOnce({
      data: {
        id: 'id',
        name: 'name',
        description: 'role',
      },
      error: undefined,
    })

    const Resolved = await resolvedComponent<Props>(AuthorContainer, {
      id: 'some id',
    })

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('name')).toBeInTheDocument()
    expect(screen.getByText('role')).toBeInTheDocument()
  })
})
