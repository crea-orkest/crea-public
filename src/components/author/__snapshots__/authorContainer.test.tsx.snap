// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author component shows all the data 1`] = `
<div>
  <div
    class="author"
  >
    <ul
      class="author__metadata"
    >
      <li
        class="author__name"
      >
        <strong>
          name
        </strong>
      </li>
      <li
        class="author__description"
      >
        role
      </li>
    </ul>
  </div>
</div>
`;
